docs/docs-beta/docs/guides/quality-testing/unit-tests-assets-and-ops.md

```diff
@@ -4,4 +4,115 @@ sidebar_position: 30
+
+<Tabs>
+  <TabItem value="asset" label="Using assets" default>
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/asset-no-argument.py" language="python"/>
+  </TabItem>
+  <TabItem value="op" label="Using ops">
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/op-no-argument.py" language="python"/>
+  </TabItem>
+</Tabs>
+
+## Testing assets and ops that have upstream dependencies
+
+If your asset or op has an upstream dependency, then you can directly pass a value for that dependency when invoking your definition.
+
+<Tabs>
+  <TabItem value="asset" label="Using assets" default>
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/asset-dependency.py" language="python" />
+  </TabItem>
+  <TabItem value="op" label="Using ops">
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/op-dependency.py" language="python" />
+  </TabItem>
+</Tabs>
+
+## Testing assets and ops with config
+
+If your asset or op uses [config](/todo), you can construct an instance of the required config object and pass it in directly.
+
+<Tabs>
+  <TabItem value="asset" label="Using assets" default>
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/asset-config.py" language="python" />
+  </TabItem>
+  <TabItem value="op" label="Using ops">
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/op-config.py" language="python" />
+  </TabItem>
+</Tabs>
+
+## Testing assets and ops with resources
+
+If your asset or op uses a [resource](/concepts/resources), it can be useful to create a mock instance of that resource to avoid interacting with external services.
+
+<Tabs>
+  <TabItem value="asset" label="Using assets" default>
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/asset-resource.py" language="python" />
+  </TabItem>
+  <TabItem value="op" label="Using ops">
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/op-resource.py" language="python" />
+  </TabItem>
+</Tabs>
+
+## Testing assets and ops with context
+
+If your asset or op uses a context argument, you can use `build_asset_context()` or `build_op_context()` to construct a context object.
+
+<Tabs>
+  <TabItem value="asset" label="Using assets" default>
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/asset-context.py" language="python" />
+  </TabItem>
+  <TabItem value="op" label="Using ops">
+    <CodeExample filePath="guides/quality-testing/unit-testing-assets-and-ops/op-context.py" language="python" />
+  </TabItem>
+</Tabs>
+
+## Testing assets and ops with a combination of parameters
+
+If your asset or op uses a combination of multiple types of parameters, it can be convenient to use keyword arguments, rather than relying on the order of parameters.
```

Comment:
I found that calling out the ordering parameters was intuitive here, perhaps because the keyword arguments were provided in their defined order.

What if instead we say something along the lines of, "When assets have multiple parameters, we recommend specifying the keyword arguments for more clarity", or something like that.

Reply:
done!